src/std/text/json/util.ss

```diff
@@ -133,9 +136,67 @@
 (defclass JSON ())
 (defmethod {:json JSON} trivial-class->json-object)
 
-(def (pretty-json object (out #f))
-  (with-output (out)
-    (filter-with-process
-     ["jq" "-M" "."]
-     (cut write-json object <>)
-     (cut copy-port <> out))))
+(def (pretty-json object (output (current-output-port))
+                  indent: (indent 2)
+                  sort-keys?: (sort-keys? (json-sort-keys))
+                  lisp-style?: (lisp-style? #f))
+  (check-argument-positive-integer indent)
+  (using (out (open-buffered-string-writer output) : BufferedStringWriter)
```

Comment:
you can also `:-` this if you care about performance, otherwise it will check method contracts.

Reply:
and even if not, you can avoid the cast with `::-`, method contracts will still be checked.

Reply:
So in the end, what does it look like to declare indent as positive fixnum in the arguments? Is it like that?
```
indent: (indent 2 ~ fx>0?)
sort-keys?: (sort-keys? (write-json-sort-keys?) : boolean::t)
lisp-style?: (lisp-style? #f : boolean::t)
```

Reply:
(indent :~ fx>0? :- :fixnum := 2)